client-docs/sync.md

```diff
@@ -0,0 +1,179 @@
+# Sync API
+
+## Table of Contents
+
+- [Description](#description)
+
+- [Types](#types)
+
+  - [`ConnectionType`](#connectiontype)
+  - [`SyncInfo`](#syncinfo)
+
+- [Methods](#methods)
+
+  - [`start`](#start)
+  - [`stop`](#stop)
+  - [`info`](#info)
+  - [`enableDiscovery`](#enablediscovery)
+  - [`disableDiscovery`](#disablediscovery)
+  - [`enableSync`](#enablesync)
+  - [`disableSync`](#disablesync)
+
+## Description
+
+Provides an imperative API for managing project discovery and sync strategies.
+
+## Types
+
+### `ConnectionType`
+
+Represents the connection type used for discovery or sync.
+
+```ts
+type ConnectionType = "lan" | "internet";
+```
+
+### `SyncInfo`
+
+Information about the kind of sync that is enabled.
+
+```ts
+type SyncInfo = {
+  discovery: ConnectionType[];
+  sync: ConnectionType[];
+};
+```
+
+## Methods
+
+### `start`
+
+`(opts: {}) => Promise<void>`
+
+Start broadcasting to peers.
+
+```ts
+// Start listening and broadcasting to other devices
+await mapeo.$sync.start({ ... })
+```
+
+**_TODO: What opts are relevant?_**
+
+**_TODO: Is it useful to return info about the subscription? (similar to `.info` return value)_**
+
+### `stop`
+
+`(opts: {}) => Promise<void>`
+
+Stop broadcasting to peers.
+
+```ts
+// Start listening and broadcasting to other devices
+await mapeo.$sync.start({ ... })
+
+// Stop listening and broadcasting
+await mapeo.$sync.stop({ ... })
+```
+
+**_TODO: Any relevant opts?_**
+**_TODO: Does it need to return anything?_**
+
+### `info`
+
+`() => Promise<SyncInfo>`
+
+Get information about discovery and sync connection types that are enabled.
+
+```ts
+// logs "{ discovery: [], sync: [] }"
+console.log(await mapeo.$sync.info());
+
+await mapeo.$sync.enableDiscovery(["internet", "lan"]);
+
+// logs "{ discovery: ["internet", "lan"], sync: [] }"
+console.log(await mapeo.$sync.info());
+```
+
+### `enableDiscovery`
+
+`(connectionTypes: ConnectionType[]) => Promise<void>`
+
+Update the discovery connection types to enable.
+
+```ts
+// Enable discovery using only internet
+await mapeo.$sync.enableDiscovery(["internet"]);
+
+console.log(await mapeo.$sync.info().discovery); // logs ["internet"]
+
+// Enable both internet and lan discovery
+await mapeo.$sync.enableDiscovery(["lan"]);
+
+console.log(await mapeo.$sync.info().discovery); // logs ["internet", "lan"]
+```
+
+**_TODO: Should `connectionTypes` override or merge existing value?_**
```

Comment:
I think it should merge, unless we decide to replace `enableX` and `disableX` with `setX` which replaces values.

Reply:
I think at some stage it will be useful to get Erik & Elys's feedback on this to see what makes most sense to them

Reply:
agree that merging makes sense if keeping the current design of enable/disable. but think I'm starting to lean towards using `set_` instead and having overriding values instead of merged ones...